pool.go

```diff
@@ -352,20 +367,32 @@ func (p *Pool[T]) Acquire(ctx context.Context) (_ *Resource[T], err error) {
 //
 // This function exists solely only for benchmarking purposes.
 func (p *Pool[T]) acquire(ctx context.Context) (*Resource[T], error) {
+	var trace tracerSpan
 	startNano := nanotime()
+	if p.tracer != nil {
+		ctx = p.tracer.AcquireStart(ctx, AcquireStartData{
+			StartNano: startNano,
+		})
```

Comment:
bundling the start call with the newtype creation might be a little too clever, since it's a minor efficiency gain at the expense of making calls to two different types, making it a little busier than in would be if the `XXXStart` methods were just on the `tracerSpan`.  Would consider reworking.